test/lib/slim-rails_assets_test.rb

```diff
@@ -28,7 +35,8 @@ class DummyApp < ::Rails::Application
 
   test 'compile slim view' do
     assert_equal 'ok', with_app(false, 'print DummyApp.assets || "ok"')
-    assert_equal '<div class="test">hi</div>', with_app(true, 'print DummyApp.assets["test.slim"].to_s')
+    assert_equal '<div class="test">hi</div>', with_app(true, 'print DummyApp.assets["test.slim"].to_s') if Sprockets::VERSION.to_i < 4
+    assert_equal '<div class="test">hi</div>', with_app(true, 'print DummyApp.assets["test.html"].to_s') if Sprockets::VERSION.to_i > 2
```

Comment:
Thanks for the PR. This part is a bit confusing to me, we have 2 lines of code, one for version < 4 (so e.g. 2 and 3) and second line for version > 2 (so 3 and 4). Both lines should be executed for v3 ?

Reply:
Correct. I was thinking to split this test

1. Base case without sprockets (first line)
2. Sprockets 2 responds with compiled template only with `slim` format (second line)
3. Sprockets 3 responds with compiled template with both `html` and `slim`formats (second and third line)
4. Sprockets 4 responds with compiled template with `html` format and with original template with `slim` format (second line + another test to get the original template)

What do you think?

Reply:
Please check if you like the new approach